ggml/src/CMakeLists.txt

```diff
@@ -490,7 +490,7 @@ if (GGML_SYCL)
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsycl-targets=nvptx64-nvidia-cuda")
         add_compile_definitions(GGML_SYCL_WARP_SIZE=32)
     else()
-        add_compile_definitions(GGML_SYCL_WARP_SIZE=32)
+        add_compile_definitions(GGML_SYCL_WARP_SIZE=16)
```

Comment:
I don't know which PR add this line to define macro GGML_SYCL_WAEP_SIZE.
But it's not good to define the macro of optimization in build script.

I suggest to mv it to source code.
There is existed function to check the Intel or other vendor.

Reply:
it can't be checked at runtime. it's a build-time macro.

Reply:
OK! Keep it!